packages/core/contracts-ovm/insured-bridge/implementation/BridgePool.sol

```diff
@@ -238,17 +265,22 @@ contract BridgePool is Testable, BridgePoolInterface, ExpandedERC20 {
 
         // Sanity check that pool has enough balance to cover relay amount + proposer reward. Reward amount will be
         // paid on settlement after the OptimisticOracle price request has passed the challenge period.
-        require(l1Token.balanceOf(address(this)) >= amount + _getProposerBond(amount), "Insufficient pool balance");
+        require(
+            l1Token.balanceOf(address(this)) >= amount + _getProposerBond(amount) &&
+                liquidReserves >= amount + _getProposerBond(amount),
+            "Insufficient pool balance"
+        );
 
         // Request a price for the relay identifier and propose "true" optimistically. These methods will pull the
-        // (proposer reward + proposer bond + final fee) from the caller.
-        // Note: We need to set a new price request timestamp instead of default setting to equal to the
-        // `depositTimestamp`, which is dependent on the L2 VM on which the DepositContract is deployed. Imagine if
-        // the timestamps on the L2 have an offset that are always "in the future" relative to L1 blocks, then the
-        // OptimisticOracle would always reject requests.
+        // (proposer reward + proposer bond + final fee) from the caller. We need to set a new price request timestamp
+        // instead of default setting to equal to the `depositTimestamp`, which is dependent on the L2 VM on which the
+        // DepositContract is deployed. Imagine if the timestamps on the L2 have an offset that are always "in the
+        // future" relative to L1 blocks, then the OptimisticOracle would always reject requests.
         _requestOraclePriceRelay(amount, priceRequestTime, getRelayAncillaryData(depositData, relayData));
         _proposeOraclePriceRelay(amount, priceRequestTime, getRelayAncillaryData(depositData, relayData));
 
+        pendingReserves += amount; // Book off maximum liquidity used by this relay in the pending reserves.
```

Comment:
Is there a reason we need to track this? If we add this to utilized reserves the moment the relay occurs (I assume we're doing this already), can we just drop this variable?

Reply:
the reason we need to track this separately(and cant use `utalizedReserves`) is that `utilizedReseres` is only incremented when the reserves are actually used. This happens when calling `settleRelay`. At this point, money from the pool is _used_. `pendingReserve` indicates funds from `liquidReserves` that are _pending_ to be used but are currently not used. I.e a relay action has _started_ (due to someone calling `relayDeposit`) but we are waiting for either a) OO liveness or b) someone to speed up the relay.

What this aims to do is prevent the case where a transfer is relayed, at which time the contract _thinks_ it has enough collateral to finish the relay and during the liveness period the LPs remove liquidity such that the relay cant be finished.

Reply:
I think this is a reasonable solution. The alternative would be to move the tokens to a separate intermediary contract, which would make accounting easier, but be more expensive gas-wise.

Reply:
You're totally right. This makes sense, and I can't think of any other way to take this into account without an additional variable. Thanks for the thorough response!!